### DISCUSSION TITLE
Deleted chats persist

### DISCUSSION THREAD

**Discussion by cyuzik** [open-webui/open-webui#5152] [General]
### Discussed in https://github.com/open-webui/open-webui/discussions/4990

<div type='discussions-op-text'>

<sup>Originally posted by **cyuzik** August 28, 2024</sup>
I was really interested in the integrated RAG and potential for internal document discussions and summarization capabilities within open-webui.

I'm very concerned that after chats are deleted, they still exist in the docker disk image in all their plain-text glory. This is a serious problem in any business that needs to delete data, especially data that has personally-identifiable-information within the chats. Anyone wanting to test this for themselves could do the following: ask a model to generate a poem about a fictional word made up of random characters such as "write a poem about 139a95e". Then when it's done, delete the chat. Then open the docker disk image using a tool such as HxD and search for 139a95e, and you'll find the chat plain as day.

I'm not sure what the best way to move forward is. Perhaps there could be some way of overwriting the chat data for the specific chat on the hard drive with random characters, then deleting but that would require the system to ensure it's overwriting the same sections of the drive.

It appears that open-webui uses SQLite as its backend database. Maybe there are ways of "vacuuming" the database after a chat delete? If not, is there some way of providing a secure delete?</div>

**Comment by tjbck** [comment]
Unable to reproduce the issue you're describing and [PostgreSQL](https://www.postgresql.org/) is supported for external DB support.

**Reply by cyuzik** [reply]
Hey tjbck,

Thanks for the response. Did you use HxD to scan the disk image for the docker container? Deleted chats persist in plain text which isn't okay, at least for our usage.

If we use an external database such as PostgreSQL does Open-WebUI tell the database to scrub or vacuum the data? Or does it at least overwrite the data with random characters before deleting the chat?

**Reply by mahenning** [reply]
Thi could be an important issue for me. @cyuzik, would it be possible for you to give a short reproducible example including the tools you use to detect that the data is not deleted? Thanks!
